packages/backend-plugin-api/src/alpha.ts

```diff
@@ -33,3 +36,32 @@ export const featureDiscoveryServiceRef =
     id: 'core.featureDiscovery',
     scope: 'root',
   });
+
+/** @alpha */
+export interface SchemaDiscoveryService {
+  getAdditionalSchemas(): Promise<{ schemas: Array<ConfigSchemaPackageEntry> }>;
```

Comment:
Do you think it would make sense to lift the abstraction up a level or two here? Instead of providing additional schemas it could instead have have this either provide the entire schema, or even the entire schema processing interface (as in `ConfigSchema['process']`. It does make the app schema loading more tricky, but then again we might need some different solution for that anyway as discussed separately.

Reply:
I'm not convinced at all it would be worth the refactoring cost.

Reply:
Doing it this way makes the addition of an external config schema provider very easy, without having to interfere with the schema loading implementation, which is a core thing.

Reply:
If you think it is needed to have this PR merged, I can even isolate the implementation of this new `alpha` service, provided in the `backend-dynamic-feature-service` package, in a dedicated `alpha` API.